tests/_test_utils/test_coordinate.py

```diff
@@ -39,156 +37,127 @@
+        """
+        Test generating lat/lon points along a great circle path.
+        """
+        lats, lons = coordinate.latlon_points(
+            ref_lats[0], ref_lons[0], ref_lats[1], ref_lons[1],
+            numpoints=numpoints, connection="greatcircle"
+        )
+        assert len(lats) == numpoints
+        assert len(lons) == numpoints
 
 
 def test_pathpoints():
+    """
+    Test generating path points with timestamps for different connections.
+    """
     lats = [0, 10]
     lons = [0, 10]
-    times = [datetime.datetime(2012, 7, 1, 10, 30),
-             datetime.datetime(2012, 7, 1, 10, 40)]
-    ref = [lats, lons, times]
-    result = coordinate.path_points(lats, lons, 100, times=times, connection="linear")
-    assert all(len(_x) == 100 for _x in result)
-    for i in range(3):
-        assert pytest.approx(result[i][0]) == ref[i][0]
-        assert pytest.approx(result[i][-1]) == ref[i][-1]
-
-    result = coordinate.path_points(lats, lons, 100, times=times, connection="greatcircle")
-    assert all(len(_x) == 100 for _x in result)
-    for i in range(3):
-        assert pytest.approx(result[i][0]) == ref[i][0]
-        assert pytest.approx(result[i][-1]) == ref[i][-1]
-
-    result = coordinate.path_points(lats, lons, 200, times=times, connection="linear")
-    assert all(len(_x) == 200 for _x in result)
-    for i in range(3):
-        assert pytest.approx(result[i][0]) == ref[i][0]
-        assert pytest.approx(result[i][-1]) == ref[i][-1]
-
-    result = coordinate.path_points(lats, lons, 200, times=times, connection="greatcircle")
-    assert all(len(_x) == 200 for _x in result)
-    for i in range(3):
-        assert pytest.approx(result[i][0]) == ref[i][0]
-        assert pytest.approx(result[i][-1]) == ref[i][-1]
-
-    lats = [0, 10, -20]
-    lons = [0, 10, 20]
-    times = [datetime.datetime(2012, 7, 1, 10, 30),
-             datetime.datetime(2012, 7, 1, 10, 40),
-             datetime.datetime(2012, 7, 1, 10, 50)]
+    times = [
+        datetime.datetime(2012, 7, 1, 10, 30),
+        datetime.datetime(2012, 7, 1, 10, 40),
+    ]
     ref = [lats, lons, times]
 
-    result = coordinate.path_points(lats, lons, 100, times=times, connection="linear")
-    assert all([len(_x) == 100 for _x in result])
-    for i in range(3):
-        assert pytest.approx(result[i][0]) == ref[i][0]
-        assert pytest.approx(result[i][-1]) == ref[i][-1]
-
-    result = coordinate.path_points(lats, lons, 100, times=times, connection="greatcircle")
-    assert all(len(_x) == 100 for _x in result)
-    for i in range(3):
-        assert pytest.approx(result[i][0]) == ref[i][0]
-        assert pytest.approx(result[i][-1]) == ref[i][-1]
+    for numpoints, connection in [(100, "linear"), (100, "greatcircle")]:
+        result = coordinate.path_points(lats, lons, numpoints, times=times, connection=connection)
+        assert all(len(_x) == numpoints for _x in result)
+        for i in range(3):
```

Comment:
From the description of eric:

https://pytest-with-eric.com/pytest-advanced/pytest-approx/#Example-Code-To-Test-Pytest-Approx

I would expect:
```
assert result[i][0] == pytest.approx(ref[i][0])
assert result[i][-1] == pytest.approx(ref[i][-1])
```

Reply:
that alone does not solve it. 

```
E       assert (-1.767766952...7669529663689) == approx((-1.76...78 ± 1.8e-06))
E         
E         comparison failed. Mismatched elements: 2 / 2:
E         Max absolute difference: 3.3047033631383727e-05
E         Max relative difference: 1.8694225263081064e-05
E         Index | Obtained            | Expected         
E         0     | -1.7677669529663687 | -1.7678 ± 1.8e-06
E         1     | 1.7677669529663689  | 1.7678 ± 1.8e-06

tests/_test_utils/test_coordinate.py:106: AssertionError
```